packages/react-diagrams-core/src/states/DragNewLinkState.ts

```diff
@@ -56,6 +60,9 @@ export class DragNewLinkState extends AbstractDisplacementState<DiagramEngine> {
 					this.link.setSourcePort(this.port);
 					this.engine.getModel().addLink(this.link);
 					this.port.reportPosition();
+
+					// save the mouse position for further precision in calculating the link's far-end point
+					this.startingPoint = new Point(event.event.clientX, event.event.clientY);
```

Comment:
The AbstractDisplacementState already contains this data in the abstract class :)

Reply:
> The AbstractDisplacementState already contains this data in the abstract class :)

that's brilliant! 
I'm going to update the PR, thank you for the tip ;)

Reply:
@dylanvorster MR updated! ;)